src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -2264,8 +2299,14 @@ private bool UseUpdatedEscapeRulesForInvocation(Symbol symbol)
                     return true;
                 }
 
+                // track the widest scope that arguments could safely escape to.
+                // use this scope as the inferred STE of declaration expressions.
+                var escapeFrom = CallingMethodScope;
```

Comment:
>escapeFrom [](http://example.com/codeflow?start=20&length=10)

The name `escapeFrom` is confusing if this is the scope that arguments can escape to.

Reply:
I think perhaps the comment should be rephrased. Definitely the name `escapeTo` is taken in this context, it refers to a destination variable whose val-escape is already known, and may be too wide to be safely used as a destination.

The goal here is to determine the widest legal destination scope for the declaration variable. This destination scope is equal to the narrowest scope of any incoming references.

Reply:
Perhaps `widestDestinationValEscape` or something like that would be a better name.